profitbricks/src/test/java/org/jclouds/profitbricks/features/DataCenterApiMockTest.java

```diff
@@ -0,0 +1,236 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+import java.util.List;
+
+import org.jclouds.profitbricks.ProfitBricksApi;
+import org.jclouds.profitbricks.domain.DataCenter;
+import org.jclouds.profitbricks.domain.Location;
+import org.jclouds.profitbricks.domain.ProvisioningState;
+import org.jclouds.profitbricks.internal.BaseProfitBricksMockTest;
+import org.jclouds.rest.ResourceNotFoundException;
+import org.testng.annotations.Test;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+/**
+ * Mock tests for the {@link org.jclouds.profitbricks.features.DataCenterApi} class
+ */
+@Test(groups = "unit", testName = "DataCenterApiMockTest")
+public class DataCenterApiMockTest extends BaseProfitBricksMockTest {
+
+   @Test
+   public void testGetAllDataCenters() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/datacenter/datacenters.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      try {
+         List<DataCenter> dataCenters = api.getAllDataCenters();
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(dataCenters);
+         assertEquals(dataCenters.size(), 2);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetDataCenter() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/datacenter/datacenter.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      String id = "12345678-abcd-efgh-ijkl-987654321000";
+      try {
+         DataCenter dataCenter = api.getDataCenter(id);
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(dataCenter);
+         assertEquals(dataCenter.id(), id);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetNonExistingContainer() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setResponseCode(500).setBody(payloadFromResource("/fault-404.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      String id = "random-non-existing-id";
+      try {
+         DataCenter dataCenter = api.getDataCenter(id);
```

Comment:
Fail the test here, otherwise the test will succeed even if the exception is not thrown.